sdk/lib/core/list.dart

```diff
@@ -464,7 +464,7 @@ abstract class List<E> implements EfficientLengthIterable<E> {
    * This increases the length of the list by one and shifts all objects
    * at or after the index towards the end of the list.
    *
-   * An error occurs if the [index] is less than 0 or greater than length.
+   * A `RangeError` occurs if the [index] is less than 0 or greater than length.
```

Comment:
I'd prefer to not say whic error is throws. Instead replace this entire section by:
```
 * The list must be growable. 
 * The [index] value must be non-negative and no greater than [length].
```

Reply:
awesome

Reply:
I think this applies to `insertAll` method as well. I am updating it for that as well.